# DADOS DO RH:

# 1.	Características do Banco de Dados:

a.	Quantas linhas e colunas temos no banco de dados?

b.	Existem dados nulos no banco? Se sim, quais colunas apresentam e quantos são os dados nulos nelas?

c.	Quais tipos de dados estão presentes no banco de dados? (Exemplo: object, int, float).

# 2.	Agrupamento de Dados e Plotagem de Gráficos:

a.	Plote um gráfico de barras (bar) que exiba a quantidade de funcionários em cada setor (coluna Department);


b.	Plote um gráfico de linha (line) que exiba a quantidade de funcionários para cada nível de escolaridade (coluna Education);


c.	Plote um gráfico de torta/pizza (pie) que exiba a quantidade de funcionários que tem ou não algum atrito na empresa (coluna Attrition).

# Importar Bibliotecas:
(vou estar usando plotly em vez de Matplotlib)

In [52]:
import pandas as pd
import plotly.express as px

# Importar Base de Dados e Visualizar:

In [53]:
df = pd.read_csv("dados_rh.csv")
display(df)

Age Attrition     BusinessTravel  DailyRate              Department  \
0      41       Yes      Travel_Rarely       1102                   Sales   
1      49        No  Travel_Frequently        279  Research & Development   
2      37       Yes      Travel_Rarely       1373  Research & Development   
3      33        No  Travel_Frequently       1392  Research & Development   
4      27        No      Travel_Rarely        591  Research & Development   
...   ...       ...                ...        ...                     ...   
1465   36        No  Travel_Frequently        884  Research & Development   
1466   39        No      Travel_Rarely        613  Research & Development   
1467   27        No      Travel_Rarely        155  Research & Development   
1468   49        No  Travel_Frequently       1023                   Sales   
1469   34        No      Travel_Rarely        628  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  \
0                    1          2  Life Sciences              1   
1                    8          1  Life Sciences              1   
2                    2          2          Other              1   
3                    3          4  Life Sciences              1   
4                    2          1        Medical              1   
...                ...        ...            ...            ...   
1465                23          2        Medical              1   
1466                 6          1        Medical              1   
1467                 4          3  Life Sciences              1   
1468                 2          3        Medical              1   
1469                 8          3        Medical              1   

      EmployeeNumber  ...  RelationshipSatisfaction StandardHours  \
0                  1  ...                         1            80   
1                  2  ...                         4            80   
2                  4  ...                         2            80   
3                  5  ...                         3            80   
4                  7  ...                         4            80   
...              ...  ...                       ...           ...   
1465            2061  ...                         3            80   
1466            2062  ...                         1            80   
1467            2064  ...                         2            80   
1468            2065  ...                         4            80   
1469            2068  ...                         1            80   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
0                    0                  8                      0   
1                    1                 10                      3   
2                    0                  7                      3   
3                    0                  8                      3   
4                    1                  6                      3   
...                ...                ...                    ...   
1465                 1                 17                      3   
1466                 1                  9                      5   
1467                 1                  6                      0   
1468                 0                 17                      3   
1469                 0                  6                      3   

     WorkLifeBalance  YearsAtCompany YearsInCurrentRole  \
0                  1               6                  4   
1                  3              10                  7   
2                  3               0                  0   
3                  3               8                  7   
4                  3               2                  2   
...              ...             ...                ...   
1465               3               5                  2   
1466               3               7                  7   
1467               3               6                  2   
1468               2               9    

Quantidade de Linhas e Colunas:

- Linhas Identificadas: 1470

- Colunas Identificadas: 35

#

# Identificar e Informar Espaços Nulos:

In [54]:
nulos_isnull = df.isnull()

print("Identificando:")
display(nulos_isnull)

colunas_vazias = df.isnull().sum().sum()
print(f"Colunas com Espaços vazios: {colunas_vazias}")

colunas_completamente_vazias = (df.isnull().sum() == len(df)).sum()
print(f"Linhas completamente vazias: {colunas_completamente_vazias}")


Identificando:


Age  Attrition  BusinessTravel  DailyRate  Department  \
0     False      False           False      False       False   
1     False      False           False      False       False   
2     False      False           False      False       False   
3     False      False           False      False       False   
4     False      False           False      False       False   
...     ...        ...             ...        ...         ...   
1465  False      False           False      False       False   
1466  False      False           False      False       False   
1467  False      False           False      False       False   
1468  False      False           False      False       False   
1469  False      False           False      False       False   

      DistanceFromHome  Education  EducationField  EmployeeCount  \
0                False      False           False          False   
1                False      False           False          False   
2                False      False           False          False   
3                False      False           False          False   
4                False      False           False          False   
...                ...        ...             ...            ...   
1465             False      False           False          False   
1466             False      False           False          False   
1467             False      False           False          False   
1468             False      False           False          False   
1469             False      False           False          False   

      EmployeeNumber  ...  RelationshipSatisfaction  StandardHours  \
0              False  ...                     False          False   
1              False  ...                     False          False   
2              False  ...                     False          False   
3              False  ...                     False          False   
4              False  ...                     False          False   
...              ...  ...                       ...            ...   
1465           False  ...                     False          False   
1466           False  ...                     False          False   
1467           False  ...                     False          False   
1468           False  ...                     False          False   
1469           False  ...                     False          False   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
0                False              False                  False   
1                False              False                  False   
2                False              False                  False   
3                False              False                  False   
4                False              False                  False   
...                ...                ...                    ...   
1465             False              False                  False   
1466             False              False                  False   
1467             False              False                  False   
1468             False              False                  False   
1469             False              False                  False   

      WorkLifeBalance  YearsAtCompany  YearsInCurrentRole  \
0               False           False               False   
1               False           False               False   
2               False           False               False   
3               False           False               False   
4               False           False               False   
...               ...             ...                 ...   
1465            False           False               False   
1466            False           False               False   
1467            False           False               False   
1468            False           False               False   
1469            False           False               False   

      YearsSinceLas

Colunas com Espaços vazios: 0
Linhas completamente vazias: 0


Nenhum Espaço Vazio Encontrado no Banco.

#

# Tipos de Dados Presentes no Banco de Dados:

In [55]:
tipos_dados = df.dtypes
display(tipos_dados)

Age                          int64
Attrition                   object
BusinessTravel              object
DailyRate                    int64
Department                  object
DistanceFromHome             int64
Education                    int64
EducationField              object
EmployeeCount                int64
EmployeeNumber               int64
EnvironmentSatisfaction      int64
Gender                      object
HourlyRate                   int64
JobInvolvement               int64
JobLevel                     int64
JobRole                     object
JobSatisfaction              int64
MaritalStatus               object
MonthlyIncome                int64
MonthlyRate                  int64
NumCompaniesWorked           int64
Over18                      object
OverTime                    object
PercentSalaryHike            int64
PerformanceRating            int64
RelationshipSatisfaction     int64
StandardHours                int64
StockOptionLevel             int64
TotalWorkingYears   

Tipos de Dados Presentes no Banco:
- object
- int64

#

# Agrupamento de Dados e Display de Gráficos:

#

### Agrupamento da Quantidade de Funcionários em cada Setor:

In [56]:
func_por_setor = df.groupby('Department')['EmployeeCount'].count().reset_index()
func_por_setor = func_por_setor.rename(columns={'EmployeeCount': 'Quantidade de Funcionarios', 'Department': 'Setor'})
display(func_por_setor)

Setor  Quantidade de Funcionarios
0         Human Resources                          63
1  Research & Development                         961
2                   Sales                         446

# Gráfico da Quantidade de Funcionários por Setor:

In [57]:
grafico_func_setor = px.bar(func_por_setor, x = 'Setor', y = 'Quantidade de Funcionarios')
grafico_func_setor.update_layout(barmode='overlay', bargap=0.6)
grafico_func_setor.update_layout(xaxis=dict(title=dict(text='Setor', font=dict(size=18))),yaxis=dict(title=dict(text='Quantidade de Funcionários', font=dict(size=18))),font=dict(size=14))
grafico_func_setor.show()

### Quantidade de Funcionários por Setor:
- Recurso Humanos: 63
- Pesquisa e Desenvolvimento: 961
- Vendas: 446

#

### Agrupamento da Quantidade de Funcionários para cada Nível de Escolaridade:

In [58]:
func_por_escola = df.groupby('Education')['EmployeeCount'].count().reset_index()
func_por_escola = func_por_escola.rename(columns={'EmployeeCount': 'Quantidade de Funcionarios', 'Education': 'Escolaridade'})
display(func_por_escola)

Escolaridade  Quantidade de Funcionarios
0             1                         170
1             2                         282
2             3                         572
3             4                         398
4             5                          48

## Gráfico da Quantidade de Funcionários por Escolaridade:

In [59]:
grafico_func_escola = px.line(func_por_escola, x = 'Escolaridade', y = 'Quantidade de Funcionarios')
grafico_func_escola.show()

### Quantidade de Funcionários por Escolaridade:

- Escolaridade 1: 170

- Escolaridade 2: 282

- Escolaridade 3: 572

- Escolaridade 4: 398

- Escolaridade 5: 48

#

### Agrupamento da Quantidade de Funcionários que Possuem ou não Atrito na Empresa:

In [60]:
func_por_atrito = df.groupby('Attrition')['EmployeeCount'].count().reset_index()
func_por_atrito = func_por_atrito.rename(columns={'EmployeeCount': 'Quantidade de Funcionarios', 'Attrition': 'Atrito'})
display(func_por_atrito)

Atrito  Quantidade de Funcionarios
0     No                        1233
1    Yes                         237

#

# Gráfico da Quantidade de Funcionários em Atrito na Empresa:

In [61]:
grafico_func_atrito = px.pie(func_por_atrito, names='Atrito', values='Quantidade de Funcionarios', title='Distribuição de Funcionários em Atrito:')
grafico_func_atrito.update_layout(uniformtext_minsize=20, uniformtext_mode='hide')
display(grafico_func_atrito)

### Quantidade de Funcionários por Atrito:

- Funcionários em Atrito: 237

- Funcionários que não estão em Atrito: 1233

#

# Fim da Análise.